mllib/src/main/scala/org/apache/spark/mllib/clustering/KMeans.scala

```diff
@@ -209,9 +211,10 @@ class KMeans private (
   /**
    * Train a K-means model on the given set of points; `data` should be cached for high
    * performance, because this is an iterative algorithm.
+   * `instr` is used to log instrumentation parameters.
    */
   @Since("0.8.0")
-  def run(data: RDD[Vector]): KMeansModel = {
+  def run(data: RDD[Vector], instr: Instrumentation[clustering.KMeans] = null): KMeansModel = {
```

Comment:
Default arguments are not Java friendly.  You'll need to do this:
```
def run(data: RDD[Vector]): KMeansModel = {
  run(data, None)
}

private[spark] def run(data: RDD[Vector], instr: Option[Instrumentation[clustering.KMeans]]): KMeansModel = ...
```
That way, we will not change the public API.  Note: I'd also use Option instead of null.